File: perl/sector-name-sic-bucket-summary.pl
use String::Similarity;


for ($i = 1; $i <= 10; $i++) {
    $file = "Sector$i-sic-freq.txt";
    print "processing $file now\n";
    open FI, $file || die "cannot open file $file";
    
    for ($k = 1; $k <= 10; $k++) {
        $c[$k] = 0;
    }

    $count = 0;
    $first = <FI>;
    while (<FI>) {
        @line = split;
#        print "$line[0], $line[1]\n";
        $two_digits = int ($line[0] / 100);
#        print "$two_digits\n";
#        <STDIN>;
        if (($two_digits > 1) && ($two_digits <= 9)) {
            $c[1] += $line[1];
        }
        elsif (($two_digits >= 10) && ($two_digits <= 14)) {
            $c[2] += $line[1];
        }
        elsif (($two_digits >= 15) && ($two_digits <= 17)) {
            $c[3] += $line[1];
        }
        elsif (($two_digits >= 20) && ($two_digits <= 39)) {
            $c[4] += $line[1];
        }
        elsif (($two_digits >= 40) && ($two_digits <= 49)) {
            $c[5] += $line[1];
        }
        elsif (($two_digits >= 50) && ($two_digits <= 51)) {
            $c[6] += $line[1];
        }
        elsif (($two_digits >= 52) && ($two_digits <= 59)) {
            $c[7] += $line[1];
        }
        elsif (($two_digits >= 60) && ($two_digits <= 67)) {
            $c[8] += $line[1];
        }
        elsif (($two_digits >= 70) && ($two_digits <= 89)) {
            $c[9] += $line[1];
        }
        elsif (($two_digits >= 91) && ($two_digits <= 99)) {
            $c[10] += $line[1];
        }
        $count++;
    }
    close FI;
    
    for ($j = 1; $j <= 10; $j++) {
        $perc = $c[$j] / $count;
#        print "SEC Sector $j: $perc\n";
        print "SEC's Sector #$j: $c[$j]\n";
    }
    print "\n";
}

File: Data/naming-evaluation/sector_name_sic_bucket_summary.py
if __name__ == '__main__':
    
    for index in range(1, 11):
        
        buckets = ((1, 9),
                   (10, 14),
                   (15, 17),
                   (20, 39),
                   (40, 49),
                   (50, 51),
                   (52, 59),
                   (60, 67),
                   (70, 89),
                   (91, 99))
        
        counts = list(0 for _ in buckets)
        
        filename = 'Sector{index}-sic-freq.txt'.format(index=index) 
        with open(filename) as f:
            for line in f:
                sic, count = line.strip().split()[:2]
                
                if sic[0] != '#':
                    sic = int(sic) // 100
                    
                    countindex = None
                    
                    for bucket, bucketkey in enumerate(buckets):
                        if bucketkey[0] <= sic <= bucketkey[1]:
                            countindex = bucket
                            
                    if countindex is not None:                            
                        counts[countindex] += int(count)
    
        print ('processing {filename}'.format(filename=filename))
        total = sum(counts)
        for countindex, count in enumerate(counts):
            print ("SEC's Sector {index}: {count} ({percentage})".format(
                index=countindex, count=count, percentage=count/total))

        print ('')
  